opt_probs.py

```diff
@@ -254,7 +254,7 @@ def eval_node_probs(self):
         # Get probs
         probs = np.zeros([self.length, self.max_val, self.max_val])
 
-        if len(self.keep_sample) == 0:      
+        if not self.keep_sample:
```

Comment:
I've noticed this change coming up in your last 2 pull requests and the reason why I don't want to go with it is because it causes an error when I run the code on my computer. I did some experimentation to try to figure out what is causing the issue and from what I can tell, `not self.keep_sample` would work if `self.keep_sample` were a list, but it doesn't work when `self.keep_sample` is a numpy array, as is the case in this code.

Perhaps, as a compromise, we could go with the following, as this is in keeping with your change and will also work with numpy arrays:
```suggestion
        if not len(self.keep_sample):
```